src/main/java/com/zerobase/cafekiosk/admin/category/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.zerobase.cafekiosk.admin.category.service;
+
+import com.zerobase.cafekiosk.admin.category.dto.CategoryDto;
+import com.zerobase.cafekiosk.admin.category.entity.Category;
+import com.zerobase.cafekiosk.admin.category.model.CategoryInput;
+import com.zerobase.cafekiosk.admin.category.repository.CategoryRepository;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+
+  private final CategoryRepository categoryRepository;
+
+  public Sort getSortBySortValue() {
+    return Sort.by(Direction.ASC, "sortValue");
+  }
+
+  @Override
+  public List<CategoryDto> list() {
+
+    List<Category> categories = categoryRepository.findAll(getSortBySortValue());
+    return CategoryDto.of(categories);
+  }
+
+  @Override
+  public boolean add(String categoryName) {
+
+    Optional<Category> optionalCategory = categoryRepository.findByCategoryName(categoryName);
+    if (optionalCategory.isPresent()) {
+      return false;
+    }
```

Comment:
optionalCategory를 별도로 선언할 이유가 없어보입니다
```suggestion
    if (categoryRepository.findByCategoryName(categoryName).isPresent()) {
      return false;
    }
```

Reply:
이미 존재하는 카테고리명으로 저장을 시도할때, 저장이 안되는데요. 사용자에게 이에 대해서 어떤 이유때문에 안되는지를 알려줄 필요는 없나요? 단순히 false로 return을 해버리면 추후 다른 비즈니스로직이 들어와서 add 가 되지 않을때도 동일하게 false, 즉 유저는 어떤 이유때문에 add가 안된것인지 알기가 어렵습니다.
보통 이런경우 exception을 던지고 GlobalExceptionHandler에서 이를 처리하도록 하면서 exception의 reason도 같이 넣어주는 형태로 개발되곤 합니다